iothub/service/src/IotHubConnectionStringBuilder.cs

```diff
@@ -17,22 +17,24 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public class IotHubConnectionStringBuilder
     {
-        const char ValuePairDelimiter = ';';
-        const char ValuePairSeparator = '=';
-        const string HostNameSeparator = ".";
-
-        static readonly string HostNamePropertyName = ((MemberExpression)((Expression<Func<IotHubConnectionStringBuilder, string>>)(_ => _.HostName)).Body).Member.Name; // todo: replace with nameof()
-        static readonly string SharedAccessKeyNamePropertyName = ((MemberExpression)((Expression<Func<IotHubConnectionStringBuilder, string>>)(_ => _.SharedAccessKeyName)).Body).Member.Name; // todo: replace with nameof()
-        static readonly string SharedAccessKeyPropertyName = ((MemberExpression)((Expression<Func<IotHubConnectionStringBuilder, string>>)(_ => _.SharedAccessKey)).Body).Member.Name; // todo: replace with nameof()
-        static readonly string SharedAccessSignaturePropertyName = ((MemberExpression)((Expression<Func<IotHubConnectionStringBuilder, string>>)(_ => _.SharedAccessSignature)).Body).Member.Name; // todo: replace with nameof();
-        static readonly Regex HostNameRegex = new Regex(@"[a-zA-Z0-9_\-\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        static readonly Regex SharedAccessKeyNameRegex = new Regex(@"^[a-zA-Z0-9_\-@\.]+$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        static readonly Regex SharedAccessKeyRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-        static readonly Regex SharedAccessSignatureRegex = new Regex(@"^.+$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
-
-        string hostName;
-        string iotHubName;
-        IAuthenticationMethod authenticationMethod;
+        private const char ValuePairDelimiter = ';';
+        private const char ValuePairSeparator = '=';
+        private const string HostNameSeparator = ".";
+
+        private static readonly string HostNamePropertyName = ((MemberExpression)((Expression<Func<IotHubConnectionStringBuilder, string>>)(_ => _.HostName)).Body).Member.Name; // todo: replace with nameof()
```

Comment:
>nameof() [](start = 199, length = 8)

Suggestion: I saw you changed to nameof in other files - maybe that could be done here with the same check in?

Reply:
Actually I don't agree with the original comment. I don't think there is any connection between the Azure ConnectionString format and the API names. They should actually be simple strings like "HostName", etc.